pkg/cmd/apply/apply.go

```diff
@@ -413,16 +415,46 @@ func (o *ApplyOptions) run(rel *apiv1.Release, storage release.Storage) (err err
 
 	// update release phase to applying
 	release.UpdateReleasePhase(rel, apiv1.ReleasePhaseApplying, relLock)
-	if err = release.UpdateApplyRelease(storage, rel, o.DryRun, relLock); err != nil {
+	if err = release.UpdateApplyRelease(releaseStorage, rel, o.DryRun, relLock); err != nil {
 		return
 	}
 
+	// Get graph storage directory, create if not exist
+	graphStorage, err := o.Backend.GraphStorage(o.RefProject.Name, o.RefWorkspace.Name)
+	if err != nil {
+		return err
+	}
+
+	// Try to get existing graph, use the graph if exists
+	if graphStorage.CheckGraphStorageExistence() {
+		gph, err = graphStorage.Get()
+		if err != nil {
+			return err
+		}
+		err = graph.ValidateGraph(gph)
+		if err != nil {
+			return err
+		}
+		// Put new resources from the generated spec to graph
+		gph, err = graph.GenerateGraph(spec.Resources, gph)
```

Comment:
This will make all the resources in the `Spec` in the status of `ApplyFailed`. If an update is interrupted, will it confuse the following three types of resources: 
- **NotApplied**
- **NotUpdated**
- **ApplyFailed**